smpplib/gsm.py

```diff
@@ -7,45 +7,15 @@
 from . import exceptions
 
 
-# from http://stackoverflow.com/questions/2452861/python-library-for-converting-plain-text-ascii-into-gsm-7-bit-character-set
-gsm = ("@£$¥èéùìòÇ\nØø\rÅåΔ_ΦΓΛΩΠΨΣΘΞ\x1bÆæßÉ !\"#¤%&'()*+,-./0123456789:;<=>"
-       "?¡ABCDEFGHIJKLMNOPQRSTUVWXYZÄÖÑÜ`¿abcdefghijklmnopqrstuvwxyzäöñüà")
-ext = ("````````````````````^```````````````````{}`````\\````````````[~]`"
-       "|````````````````````````````````````€``````````````````````````")
-if six.PY2:
-    gsm = gsm.decode('utf-8')
-    ext = ext.decode('utf-8')
-
-
-class EncodeError(ValueError):
-    """Raised if text cannot be represented in gsm 7-bit encoding"""
-
-
-def gsm_encode(plaintext, hex=False):
-    """Replace non-GSM ASCII symbols"""
-    res = ""
-    for c in plaintext:
-        idx = gsm.find(c)
-        if idx != -1:
-            res += chr(idx)
-            continue
-        idx = ext.find(c)
-        if idx != -1:
-            res += chr(27) + chr(idx)
-            continue
-        raise EncodeError()
-    return binascii.b2a_hex(res) if hex else res
-
-
 def make_parts(text):
     """Returns tuple(parts, encoding, esm_class)"""
     try:
-        text = gsm_encode(text)
-        encoding = consts.SMPP_ENCODING_DEFAULT
-        need_split = len(text) > consts.SEVENBIT_SIZE
-        partsize = consts.SEVENBIT_MP_SIZE
-        encode = six.b
-    except EncodeError:
+        text = text.encode('iso-8859-1')
+        encoding = consts.SMPP_ENCODING_ISO88591
+        need_split = len(text) > consts.EIGHTBIT_SIZE
+        partsize = consts.EIGHTBIT_MP_SIZE
+        encode = lambda s: s
```

Comment:
are you sure it will work in both python2 and python3?

Reply:
It should work. Anyway, I'll change that because of additional parameter.